internal/http/middleware/status.go

```diff
@@ -0,0 +1,31 @@
+package middleware
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/BoilerMake/new-backend/internal/status"
+)
+
+// OnSeasonOnly make sure the APP_STATUS environment variable is between 2 and 4, if
+// it's outside of that then
+func OnSeasonOnly(h http.Handler) http.Handler {
+	statusString := mustGetEnv("APP_STATUS")
+	statusInt, err := strconv.Atoi(statusString)
+	if err != nil {
+		log.Fatalf("Failed to convert status to int: %v", err)
+	}
+	st := status.Status(statusInt)
```

Comment:
Is it possible to do the string to int conversion within the Status() method so you don't have to do it before every time you call it?

Reply:
That's actually just a type conversion so the variable has a bit of semantic meaning (it's of type Status instead of int because it specifically represents a status).  

Also, this is only called once during startup, the middleware returns a closure function which keeps the `st` variable in scope but doesn't have to get it or recalculate it every time a request goes through this middleware.

Also, this made me realize I forgot to actually use the Status like a Status instead of just like an int, I'll make it do that in the next commit